src/modules/extrakeys/main_impl.h

```diff
@@ -55,6 +55,19 @@ int secp256k1_xonly_pubkey_serialize(const secp256k1_context* ctx, unsigned char
     return 1;
 }
 
+int secp256k1_xonly_pubkey_cmp(const secp256k1_context* ctx, const secp256k1_xonly_pubkey* pk1, const secp256k1_xonly_pubkey* pk2) {
+    unsigned char out1[32];
+    unsigned char out2[32];
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(pk1 != NULL);
+    ARG_CHECK(pk2 != NULL);
```

Comment:
`ARG_CHECK` returns 0 for failure here, which isn't an appropriate return condition for a `cmp` function.  You can use `ARG_CHECK_NO_RETURN`, but you won't have a post-condition that the pointer is non-NULL, and you will still need to handle that case.  This can be done by zeroing the `out1` and `out2` buffers and only calling `secp256k1_xonly_pubkey_serialize` when `pk` is non NULL.